### PR TITLE
Out of memory errors

### PR DESCRIPTION
**Pull request by PiotrekZak**
Should be merged after feat/geo_gaps

### CODE REVIEW

File: zrp/zrp.py [hunk 1]
```diff
@@ -96,14 +96,45 @@ def rename_data_columns(self, data):
         # self.params_dict = {}
         return data
     
-    def transform(self, input_data):
+    def delete_old_files(self):
```
Comment (kasey-zest) [inline zrp/zrp.py:99]: Could lead to accidental deletions, adding a message warning the user to provide a new directory path might be better than file deletion.

File: zrp/prepare/geo_lookup.py [hunk 2]
```diff
@@ -113,11 +114,12 @@ class GeoLookUpBuilder():
         Year associated with ACS data. 
     """
 
-    def __init__(self, support_files_path, year):
+    def __init__(self, support_files_path, year, output_folder_suffix):
```
Comment (kasey-zest) [inline zrp/prepare/geo_lookup.py:117]: `output_folder_suffix` missing from docstring

File: zrp/zrp.py (hunk 1 above)
Comment (PiotrekZak) [reply zrp/zrp.py]: Good idea – I changed the code as suggested

**Review by kasey-zest — APPROVED** [review]

File: zrp/prepare/geo_geocoder.py [hunk 3]
```diff
@@ -101,85 +65,43 @@ class ZGeo(BaseZRP):
...
+        Checks if a number is odd.
         
-        return(geo_df)    
+        Parameters
+        ----------
+        s: string
+            String with a number to be checked
+        """
+        try:
+            return (int(str(s)[-1]) % 2)
```
[the remark this answers is not in the document]
Comment (PiotrekZak) [reply zrp/prepare/geo_geocoder.py:102]: The numeric portion of the string should be the only part that is evaluated as odd or even and this is how it is done in the code. In the lines below we take only the "right" part of the house numbers - "RIGHT" is numeric and "left" is non-numeric.

geo_df["Parity_Match"] = np.where(
            (geo_df.FROMHN_RIGHT.apply(self.__is_odd)) == (geo_df.house_number_RIGHT.apply(self.__is_odd)) ,
            1,
            0)

**Pull request merged**